### DISCUSSION TITLE
[4.x]: Matrix - Propagation method - Blocks not cloned

### DISCUSSION THREAD

**Discussion by JeanLucEsser** [craftcms/cms#12203] [Ideas]
### Description

First things first, this is for a section whose propagation method is set to `Let each entry choose which site it should be saved to`. The entry in this section has a Matrix field whose propagation method is `Only save blocks to the site they were created in`.

I liken the Matrix propagation method to the translation method of more classic fields. In the stated configuration, adding a new site to the entry would clone the content of every translated field (and obviously not translated ones as well).

I would expect the same for the Matrix blocks, clone them when adding a new site and let them live their life after that. As it stands, the Matrix field starts fresh and empty.

### Steps to reproduce

1. Create a section with `Let each entry choose which site it should be saved to`
2. Add a Matrix field with `Only save blocks to the site they were created in`
3. Create a new entry with a few matrix blocks and save it
4. Add a site to this entry

### Expected behavior

The Matrix field should be cloned and then let me remove and/or add blocks

### Actual behavior

The Matrix field is emptied, all content within the blocks is lost

### Craft CMS version

4.3.0

**Comment by JeanLucEsser** [comment]
More thinking on the consequences of this:

I guess in the described case (`Only save blocks to the site they were created in`) every block (even when cloned if we get there) has its own id. Which would then make the option of translating block fields something that shouldn't have any effect. Wouldn't it make sense that the option of translating block fields be disabled when this propagation method is set?

The translated icon should not be shown at the field level either but if shown at all it should be at the block level.

**Reply by brandonkelly** [reply]
Moved this to a discussion as it’s currently working as intended.

**Comment by JeanLucEsser** [comment]
Fair enough. So my argument is simple here.

Why wouldn't matrix blocks be cloned when classic translatable fields are.

Or put another way:
Why would the only way of cloning the content of a matrix be at the expense of having to keep the blocks structure the same.

Matrixes are just a way of repeating fields, organized in blocks. If a translatable field outside a matrix can be cloned, a translatable field inside a matrix should be as well, wether we have the intent of propagating the blocks structure or not. And the only way of allowing for that is to clone the blocks even when we do not want to propagate them.

Going further, the fields inside a block should only be translatable if that block is propagated in some way. If it never is, by definition, the block is unique, and so are the fields content inside it making the translation option useless.
